lightning-net-tokio/src/lib.rs

```diff
@@ -157,14 +157,20 @@ impl Connection {
 	}
 }
 
-#[derive(Clone)]
+/// SimpleArcPeerManager defines a type alias for the peer_hander's SimpleArcPeerManager, with
+/// a concrete type given to its Descriptor generic parameter. We need to use the
+/// peer_hander::SimpleArcPeerManager instead of the peer_hander::SimpleRefPeerManager because
+/// we're using this parameter in calls to tokio::spawn, which requires parameters to have
+/// static lifetimes. Aliasing this concrete type allows for more concise function definitions.
+pub type SimpleArcPeerManager = peer_handler::SimpleArcPeerManager<SocketDescriptor>;
+
 pub struct SocketDescriptor {
 	conn: Arc<Mutex<Connection>>,
 	id: u64,
-	peer_manager: Arc<peer_handler::PeerManager<SocketDescriptor>>,
+	peer_manager: SimpleArcPeerManager,
```

Comment:
Hmm, doesn't this restrict us to very specific use-cases eg InMemoryChannelKeys + SimpleManyChannelMonitor<OutPoint>? I think we may need to template + default these too :(.

Reply:
I reverted back to the previous version. This makes it "templated" iiuc, but not sure what
> \+ default

means :sweat_smile: Soz if missing the obvious here..